coalib/results/TextRange.py

```diff
@@ -89,7 +89,35 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        first = False  # check for left boundary overlap
+        # self.start <= other.end
+
+        second = False  # check for right boundary overlap
+        # self.end >= other.start
+
+        if None not in (self.start.line, other.end.line):
```

Comment:
I would refactor to `if self.start.line is not None and other.end.line is not None` -> faster and imo slightly better to understand.

But up to you :+1: